pkg/certificate/interface.go

```diff
@@ -0,0 +1,101 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	Auto        ProviderType = "auto"
+	CertManager ProviderType = "cert-manager"
+	// add more ...
+)
+
+func NewProvider(pt ProviderType) Provider {
+	switch pt {
+	case Auto:
+		return nil // change me later
+	case CertManager:
+		return nil // change me later
+	}
+	panic(fmt.Sprintf("unknown provider type: %s", pt))
```

Comment:
better to return error instead of panic.

Reply:
updated & thx